Index: lib/private/Template/CSSResourceLocator.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- lib/private/Template/CSSResourceLocator.php	(revision 4f2837017c95f063d99a05b6cd6a5e73ac388d75)
+++ lib/private/Template/CSSResourceLocator.php	(date 1574619791911)
@@ -125,26 +125,7 @@
 		if (!$scss) {
 			parent::append($root, $file, $webRoot, $throw);
 		} else {
-			if (!$webRoot) {
-				$webRoot = $this->findWebRoot($root);
-
-				if ($webRoot === null) {
-					$webRoot = '';
-					$this->logger->error('ResourceLocator can not find a web root (root: {root}, file: {file}, webRoot: {webRoot}, throw: {throw})', [
-						'app' => 'lib',
-						'root' => $root,
-						'file' => $file,
-						'webRoot' => $webRoot,
-						'throw' => $throw ? 'true' : 'false'
-					]);
-
-					if ($throw && $root === '/') {
-						throw new ResourceNotFoundException($file, $webRoot);
-					}
-				}
-			}
-
-			$this->resources[] = array($webRoot? : \OC::$WEBROOT, $webRoot, $file);
+			$this->resources[] = array(\OC::$WEBROOT, \OC::$WEBROOT, $file);
 		}
 	}
 }
