diff --git a/src/libical/icaltz-util.c b/src/libical/icaltz-util.c
index 982d8a74..0993c0d9 100644
--- a/src/libical/icaltz-util.c
+++ b/src/libical/icaltz-util.c
@@ -97,9 +97,9 @@ static char s_zoneinfopath[MAXPATHLEN] = {0};
 
 /* A few well-known locations for system zoneinfo; can be overridden with TZDIR environment */
 static const char *s_zoneinfo_search_paths[] = {
+    "/etc/zoneinfo",
     "/usr/share/zoneinfo",
     "/usr/lib/zoneinfo",
-    "/etc/zoneinfo",
     "/usr/share/lib/zoneinfo"
 };
 
