--- a/meson.build
+++ b/meson.build
@@ -909,11 +909,11 @@ meson.add_install_script(
   join_paths('tools', 'meson-post-install.sh'),
   nm_datadir,
   nm_bindir,
-  nm_pkgconfdir,
+  nm_prefix + nm_pkgconfdir,
   nm_pkglibdir,
-  nm_pkgstatedir,
+  nm_prefix + nm_pkgstatedir,
   nm_mandir,
-  nm_sysconfdir,
+  nm_prefix + nm_sysconfdir,
   enable_docs ? '1' : '0',
   enable_ifcfg_rh ? '1' : '0',
   enable_nm_cloud_setup ? '1' : '0',
