--- a/meson.build
+++ b/meson.build
@@ -925,11 +925,11 @@
   join_paths('tools', 'meson-post-install.sh'),
   nm_datadir,
   nm_bindir,
-  nm_pkgconfdir,
+  nm_prefix + nm_pkgconfdir,
   nm_pkglibdir,
-  nm_pkgstatedir,
+  nm_prefix + nm_pkgstatedir,
   enable_docs ? 'install_docs' : '',
   nm_mandir,
   enable_ifcfg_rh ? 'create_network_scripts' : '',
-  nm_sysconfdir,
+  nm_prefix + nm_sysconfdir,
 )
