From c2fdc49827aaf7f5460bbd419ad7b8bbec616f7e Mon Sep 17 00:00:00 2001
From: Will Dietz <w@wdtz.org>
Date: Mon, 7 Oct 2019 20:08:23 -0500
Subject: [PATCH] bluetooth/meson.build: fix build when tests are disabled

---
 src/devices/bluetooth/meson.build | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/devices/bluetooth/meson.build b/src/devices/bluetooth/meson.build
index 63c47bcba..ec77650f2 100644
--- a/src/devices/bluetooth/meson.build
+++ b/src/devices/bluetooth/meson.build
@@ -35,6 +35,7 @@ test(
   args: [libnm_device_plugin_bluetooth.full_path(), linker_script_devices],
 )
 
+if enable_tests
 test_unit = 'nm-bt-test'
 
 executable(
@@ -43,3 +44,4 @@ executable(
   dependencies: libnetwork_manager_test_dep,
   c_args: test_c_flags,
 )
+endif
-- 
2.23.GIT

