diff --git a/src/devices/wifi/nm-iwd-manager.c b/src/devices/wifi/nm-iwd-manager.c
index dd1cad480..039ccb44d 100644
--- a/src/devices/wifi/nm-iwd-manager.c
+++ b/src/devices/wifi/nm-iwd-manager.c
@@ -257,7 +257,7 @@ register_agent (NMIwdManager *self)
 	GDBusInterface *agent_manager;
 
 	agent_manager = g_dbus_object_manager_get_interface (priv->object_manager,
-	                                                     "/",
+	                                                     NM_IWD_PATH,
 	                                                     NM_IWD_AGENT_MANAGER_INTERFACE);
 
 	/* Register our agent */
diff --git a/src/devices/wifi/nm-iwd-manager.h b/src/devices/wifi/nm-iwd-manager.h
index c50963fe1..ffc1ee17b 100644
--- a/src/devices/wifi/nm-iwd-manager.h
+++ b/src/devices/wifi/nm-iwd-manager.h
@@ -11,6 +11,7 @@
 
 #define NM_IWD_BUS_TYPE                 G_BUS_TYPE_SYSTEM
 #define NM_IWD_SERVICE                  "net.connman.iwd"
+#define NM_IWD_PATH                     "/net/connman/iwd"
 
 #define NM_IWD_AGENT_MANAGER_INTERFACE  "net.connman.iwd.AgentManager"
 #define NM_IWD_WIPHY_INTERFACE          "net.connman.iwd.Adapter"
