diff --git a/certbot/setup.py b/certbot/setup.py
index 0026ef8e9..6835a6263 100644
--- a/certbot/setup.py
+++ b/certbot/setup.py
@@ -1,5 +1,4 @@
 import codecs
-from distutils.version import StrictVersion
 import os
 import re
 import sys
@@ -58,20 +57,8 @@ install_requires = [
 
 # Add pywin32 on Windows platforms to handle low-level system calls.
 # This dependency needs to be added using environment markers to avoid its installation on Linux.
-# However environment markers are supported only with setuptools >= 36.2.
-# So this dependency is not added for old Linux distributions with old setuptools,
-# in order to allow these systems to build certbot from sources.
 pywin32_req = 'pywin32>=227'  # do not forget to edit pywin32 dependency accordingly in windows-installer/construct.py
-if StrictVersion(setuptools_version) >= StrictVersion('36.2'):
-    install_requires.append(pywin32_req + " ; sys_platform == 'win32'")
-elif 'bdist_wheel' in sys.argv[1:]:
-    raise RuntimeError('Error, you are trying to build certbot wheels using an old version '
-                       'of setuptools. Version 36.2+ of setuptools is required.')
-elif os.name == 'nt':
-    # This branch exists to improve this package's behavior on Windows. Without
-    # it, if the sdist is installed on Windows with an old version of
-    # setuptools, pywin32 will not be specified as a dependency.
-    install_requires.append(pywin32_req)
+install_requires.append(pywin32_req + " ; sys_platform == 'win32'")
 
 dev_extras = [
     'coverage',
