From e4bfc845258252dbf269467c6e0c2789ad9f30b4 Mon Sep 17 00:00:00 2001
From: Will Dietz <w@wdtz.org>
Date: Mon, 3 Feb 2020 06:54:25 -0600
Subject: [PATCH] don't force elementary themes

---
 src/Application.vala | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/Application.vala b/src/Application.vala
index 43f0613..d6555fe 100644
--- a/src/Application.vala
+++ b/src/Application.vala
@@ -116,8 +116,8 @@ public class Planner : Gtk.Application {
 
         utils.apply_theme_changed ();
 
-        Gtk.Settings.get_default ().set_property ("gtk-icon-theme-name", "elementary");
-        Gtk.Settings.get_default ().set_property ("gtk-theme-name", "elementary");
+        // Gtk.Settings.get_default ().set_property ("gtk-icon-theme-name", "elementary");
+        // Gtk.Settings.get_default ().set_property ("gtk-theme-name", "elementary");
 
         // Set shortcut
         utils.set_quick_add_shortcut ("<Alt>a");
-- 
2.25.0.2

