diff --git a/QtSpim.pro b/QtSpim.pro2
index d81b2b0..d0de616 100644
--- a/QtSpim.pro
+++ b/QtSpim.pro2
@@ -105,31 +105,6 @@ QMAKE_LEXFLAGS_MANGLE = -Pyy
 QMAKE_LEXFLAGS      = -I -8 --outfile=lex.scanner.c
 
 
-# Help file
-#
-HELP_PROJ           = help/qtspim.qhp
-buildcompressedhelp.name    = Build compressed help
-buildcompressedhelp.input   = HELP_PROJ
-buildcompressedhelp.output  = help/${QMAKE_FILE_BASE}.qch
-buildcompressedhelp.commands= qhelpgenerator ${QMAKE_FILE_IN} -o ${QMAKE_FILE_OUT}
-buildcompressedhelp.CONFIG  = no_link recursive
-
-# qcollectiongenerator must be run in the directory containing the project file, otherwise it
-# puts partial paths in the .qhc file, which make it impossible to install the help files in
-# other directories.
-#
-HELP_COL_PROJ       = help/qtspim.qhcp
-buildhelpcollection.name    = Build help collection
-buildhelpcollection.input   = HELP_COL_PROJ
-buildhelpcollection.output  = help/${QMAKE_FILE_BASE}.qhc
-linux|macx:buildhelpcollection.commands= bash -c '\"pushd ${QMAKE_FILE_PATH}; qcollectiongenerator ${QMAKE_FILE_BASE}.qhcp; popd; $(MOVE) ${QMAKE_FILE_PATH}/${QMAKE_FILE_BASE}.qhc ${QMAKE_FILE_OUT};\"'
-win32:buildhelpcollection.commands= cmd -c '\"pushd ${QMAKE_FILE_PATH} & qcollectiongenerator ${QMAKE_FILE_BASE}.qhcp & popd & $(MOVE) ${QMAKE_FILE_PATH}\\${QMAKE_FILE_BASE}.qhc ${QMAKE_FILE_OUT}\"'
-buildhelpcollection.CONFIG  = no_link recursive
-
-QMAKE_EXTRA_COMPILERS       += buildcompressedhelp buildhelpcollection
-POST_TARGETDEPS             += help/qtspim.qch help/qtspim.qhc
-
-
 # Microsoft Visual C compiler flags
 #
 win32-msvc2008 {
