From 13b17c9c9219de819e0b3ba7e52d75b0861c69e9 Mon Sep 17 00:00:00 2001
From: Will Dietz <w@wdtz.org>
Date: Fri, 2 Aug 2019 17:21:55 -0500
Subject: [PATCH] handle reports w/o columns info. "timesheets" for me doesn't.

---
 vit/config_parser.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/vit/config_parser.py b/vit/config_parser.py
index 08fa7a5..92285b7 100644
--- a/vit/config_parser.py
+++ b/vit/config_parser.py
@@ -343,6 +343,8 @@ class TaskParser(object):
         return primary_sort[0] == 'project' and primary_sort[1] == 'ascending'
 
     def get_column_index(self, report_name, column):
+        if not 'columns' in self.reports[report_name]:
+            return None
         return self.reports[report_name]['columns'].index(column) if column in self.reports[report_name]['columns'] else None
 
     def get_column_label(self, report_name, column):
-- 
2.22.0

