From addb13a1542188acf87be0dc856bf06fde3e4a67 Mon Sep 17 00:00:00 2001
From: Will Dietz <w@wdtz.org>
Date: Mon, 17 Feb 2020 23:04:59 -0600
Subject: [PATCH] manually restore l_ascii_table

---
 ell/ell.sym | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ell/ell.sym b/ell/ell.sym
index 27faefb..766c239 100644
--- a/ell/ell.sym
+++ b/ell/ell.sym
@@ -531,6 +531,7 @@ global:
 	l_uintset_put;
 	l_uintset_take;
 	/* utf8 */
+	l_ascii_table;
 	l_utf8_from_ucs2be;
 	l_utf8_from_utf16;
 	l_utf8_from_wchar;
-- 
2.25.1

