From 668245c3c321d8a5d90d9b7a69f353de1718a762 Mon Sep 17 00:00:00 2001
From: Will Dietz <w@wdtz.org>
Date: Mon, 6 Apr 2020 10:25:29 -0500
Subject: [PATCH] add l_genl_msg_get_extended_error to exported syms

---
 ell/ell.sym | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ell/ell.sym b/ell/ell.sym
index f4b30dd..b718264 100644
--- a/ell/ell.sym
+++ b/ell/ell.sym
@@ -265,6 +265,7 @@ global:
 	l_genl_msg_get_command;
 	l_genl_msg_get_version;
 	l_genl_msg_get_error;
+	l_genl_msg_get_extended_error;
 	l_genl_msg_append_attr;
 	l_genl_msg_append_attrv;
 	l_genl_msg_enter_nested;
-- 
2.26.0

