From 00fdde6f46b67f500d7fbfa67909a9304accdacd Mon Sep 17 00:00:00 2001
From: Will Dietz <w@wdtz.org>
Date: Wed, 23 Oct 2019 09:04:01 -0500
Subject: [PATCH 2/4] build-presets: (linux) allow custom install prefix

---
 utils/build-presets.ini | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/utils/build-presets.ini b/utils/build-presets.ini
index a9e15c9e27..348bf6f2de 100644
--- a/utils/build-presets.ini
+++ b/utils/build-presets.ini
@@ -752,7 +752,7 @@ install-llbuild
 install-swiftpm
 install-xctest
 install-libicu
-install-prefix=/usr
+install-prefix=%(install_prefix)s
 swift-install-components=autolink-driver;compiler;clang-resource-dir-symlink;stdlib;swift-remote-mirror;sdk-overlay;parser-lib;toolchain-tools;license;sourcekit-inproc
 llvm-install-components=llvm-cov;llvm-profdata;IndexStore;clang;clang-headers;compiler-rt;clangd
 install-libcxx
-- 
2.24.0-rc0

