diff --git a/lib/Driver/ToolChains/CommonArgs.cpp b/lib/Driver/ToolChains/CommonArgs.cpp
index 6b6e276b8ce7..7896542a1202 100644
--- a/lib/Driver/ToolChains/CommonArgs.cpp
+++ b/lib/Driver/ToolChains/CommonArgs.cpp
@@ -409,9 +409,7 @@ void tools::addLTOOptions(const ToolChain &ToolChain, const ArgList &Args,
 
     SmallString<1024> Plugin;
-    llvm::sys::path::native(Twine(D.Dir) +
-                                "/../" CLANG_INSTALL_LIBDIR_BASENAME +
-                                PluginName + Suffix,
+    llvm::sys::path::native(Twine("@libllvmLibdir@") + PluginName + Suffix,
                             Plugin);
     CmdArgs.push_back(Args.MakeArgString(Twine(PluginPrefix) + Plugin));
          Plugin);
   }
