From 15579d957e1c79fe591954bb050451680c4c3aa1 Mon Sep 17 00:00:00 2001
From: Will Dietz <w@wdtz.org>
Date: Tue, 20 Aug 2019 19:56:50 -0500
Subject: [PATCH] src/manager.c: fix build, use l_genl_msg_new_sized

---
 src/manager.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/manager.c b/src/manager.c
index 653b57f..8a7e434 100644
--- a/src/manager.c
+++ b/src/manager.c
@@ -722,7 +722,7 @@ bool manager_init(struct l_genl_family *in,
 		return false;
 	}
 
-	msg = l_genl_msg_new(NL80211_CMD_GET_WIPHY, 128);
+	msg = l_genl_msg_new_sized(NL80211_CMD_GET_WIPHY, 128);
 	l_genl_msg_append_attr(msg, NL80211_ATTR_SPLIT_WIPHY_DUMP, 0, NULL);
 	wiphy_dump = l_genl_family_dump(nl80211, msg,
 						manager_wiphy_dump_callback,
-- 
2.23.0

