From 992b3c5fe97cc88a8c21558cbd2d6fb843a25688 Mon Sep 17 00:00:00 2001
From: Will Dietz <w@wdtz.org>
Date: Mon, 4 Nov 2019 16:48:23 -0600
Subject: [PATCH] unit/test-wsc.c: fix inverted assert condition

---
 unit/test-wsc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/unit/test-wsc.c b/unit/test-wsc.c
index 0e845dc2..b329c9ee 100644
--- a/unit/test-wsc.c
+++ b/unit/test-wsc.c
@@ -1948,7 +1948,7 @@ static void verify_handshake_event(struct handshake_state *hs,
 	{
 		const struct wsc_credential *cred;
 
-		assert(va_arg(args, unsigned int) !=
+		assert(va_arg(args, unsigned int) ==
 			EAP_WSC_EVENT_CREDENTIAL_OBTAINED);
 
 		cred = va_arg(args, const struct wsc_credential *);
-- 
2.24.0

