From dd8bf4e33654d807947217e284d5ab8eba73a244 Mon Sep 17 00:00:00 2001
From: Will Dietz <w@wdtz.org>
Date: Wed, 25 Sep 2019 14:28:11 -0500
Subject: [PATCH] Handle lack of /dev/tty, fix test.

---
 khal/cli.py | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/khal/cli.py b/khal/cli.py
index b2354c2..a79eb4a 100644
--- a/khal/cli.py
+++ b/khal/cli.py
@@ -438,10 +438,13 @@ def _get_cli():
             if not ics:
                 ics_strs = (sys.stdin.read(),)
                 if not batch:
-                    if os.stat('/dev/tty').st_mode & stat.S_IFCHR > 0:
-                        sys.stdin = open('/dev/tty', 'r')
-                    else:
+                    try:
+                        if os.stat(os.ctermid()).st_mode & stat.S_IFCHR > 0:
+                            sys.stdin = open(os.ctermid(), 'r')
+                    except FileNotFoundError:
                         logger.warning('/dev/tty does not exist, importing might not work')
+                    except OSError:
+                        pass
             else:
                 ics_strs = (ics_file.read() for ics_file in ics)
 
-- 
2.23.GIT

