diff --git a/scripts/Makefile.modinst b/scripts/Makefile.modinst
index 07650ee..934a7a8 100644
--- a/scripts/Makefile.modinst
+++ b/scripts/Makefile.modinst
@@ -9,3 +9,4 @@ include scripts/Kbuild.include
 include scripts/Kbuild.include

-modules := $(sort $(shell cat $(if $(KBUILD_EXTMOD),$(KBUILD_EXTMOD)/)modules.order))
+modules := $(sort $(foreach f,$(wildcard $(MODVERDIR)/*.mod),$(shell \
+    grep -h '\.ko$$' '$f')))

 PHONY += $(modules)
